### PR TITLE
Update: Add unique validation for organizational metadata on namespace

### PR DESCRIPTION
**Pull request by ericrpowers**
#### Description
This adds a check on all organizational metadata on a namespace to have a specific prefix (currently `@om:` but can be changed), expands ValidateMetadata to check no basic metadata has this prefix, and removes an unnecessary validation on specs/organizationalmetadata.spec as this is validated by the namespace already.

### CODE REVIEW

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "org-meta-hooks",
    "component": "gaia",
    "pr-id": "477",
    "commit-sha": "240f117d076b6c5a6f026139910a5a6d697d5519",
    "pipeline": "master"
  },
  {
    "project": "org-meta-hooks",
    "component": "backend",
    "pr-id": "730",
    "commit-sha": "8e033227597e1625583404915c3bb738bf4cb5da",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "org-meta-hooks",
    "component": "backend",
    "pr-id": "730",
    "commit-sha": "5b124dbf80c24e84ff9647d4028b401af4ff4336",
    "pipeline": "master"
  },
  {
    "project": "org-meta-hooks",
    "component": "gaia",
    "pr-id": "477",
    "commit-sha": "240f117d076b6c5a6f026139910a5a6d697d5519",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "org-meta-hooks",
    "component": "backend",
    "pr-id": "730",
    "commit-sha": "7a09f6f77549730ddd029a2ef15ab54746b8e466",
    "pipeline": "master"
  },
  {
    "project": "org-meta-hooks",
    "component": "gaia",
    "pr-id": "477",
    "commit-sha": "e1f41bb2d26fbbab6c386dff42ea08b89290c586",
    "pipeline": "master"
  }
]
```

File: custom_validations.go [hunk 1]
```diff
@@ -760,17 +760,39 @@ func ValidateTagsExpression(attribute string, expression [][]string) error {
...
 
-// ValidateMetadata validates an []string is a valid list of metadata.
+// ValidateMetadata validates if a []string is a valid list of metadata.
 func ValidateMetadata(attribute string, metadata []string) error {
 
 	for _, m := range metadata {
 
 		if !strings.HasPrefix(m, prefixMetadata) {
-			return makeValidationError(attribute, fmt.Sprintf("Metadata %s does not starts with an @, a $ or a + that is reserved", m))
+			return makeValidationError(attribute, fmt.Sprintf("Metadata %s does not start with an @, a $ or a +", m))
```
Comment (primalmotion) [inline custom_validations.go:769]: this message is less wrong that is was before, but it's sill is.

It should be `Metadata %s must start with an @`

File: constants/constants.go [hunk 2]
```diff
@@ -370,6 +370,13 @@ const (
 	MetadataUsrPortKey = "@usr:port"
 )
 
+// Organizational metadata tag constants.
+const (
+
+	// OrganizationalMetadataKey is the key of an organizatonal metadata.
+	OrganizationalMetadataKey = "@om:"
```
Comment (t00f) [inline constants/constants.go:377]: Prefix should be `@org:`

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "org-meta-hooks",
    "component": "backend",
    "pr-id": "730",
    "commit-sha": "a3e70354e23137e7537d62cd95a211cdbec618e9",
    "pipeline": "master"
  },
  {
    "project": "org-meta-hooks",
    "component": "gaia",
    "pr-id": "477",
    "commit-sha": "dc2eb1102e46218db1f9ebb17513de4dfd4fd07a",
    "pipeline": "master"
  }
]
```

File: custom_validations.go [hunk 3]
```diff
@@ -760,17 +760,39 @@ func ValidateTagsExpression(attribute string, expression [][]string) error {
...
 
-// ValidateMetadata validates an []string is a valid list of metadata.
+// ValidateMetadata validates if a []string is a valid list of metadata.
 func ValidateMetadata(attribute string, metadata []string) error {
 
 	for _, m := range metadata {
 
 		if !strings.HasPrefix(m, prefixMetadata) {
-			return makeValidationError(attribute, fmt.Sprintf("Metadata %s does not starts with an @, a $ or a + that is reserved", m))
+			return makeValidationError(attribute, fmt.Sprintf("Metadata %s must start with an @, a $ or a +", m))
```
Comment (primalmotion) [inline custom_validations.go:769]: this is still wrong: `"Metadata %s must start with an @"`

File: custom_validations.go [hunk 4]
```diff
@@ -760,17 +760,39 @@ func ValidateTagsExpression(attribute string, expression [][]string) error {
...
 		}
 
 		if strings.HasPrefix(m, constants.AuthKey) {
-			return makeValidationError(attribute, fmt.Sprintf("Metadata %s is using @auth: which is reserved", m))
+			return makeValidationError(attribute, fmt.Sprintf("Metadata %s is using %s which is reserved", m, constants.AuthKey))
+		}
+
+		if strings.HasPrefix(m, constants.OrganizationalMetadataKey) {
```
Comment (primalmotion) [inline custom_validations.go:776]: `"Metadata %s must not start with the reserved prefix %s`

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "org-meta-hooks",
    "component": "backend",
    "pr-id": "730",
    "commit-sha": "e9332a5063caedb641035c365f707859bd8ab5fb",
    "pipeline": "master"
  },
  {
    "project": "org-meta-hooks",
    "component": "gaia",
    "pr-id": "477",
    "commit-sha": "ed2c269b280a1abfebd04b92940dc2a1356cc86d",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "org-meta-hooks",
    "component": "backend",
    "pr-id": "730",
    "commit-sha": "1594990cdeeed6bc09fc9db9e57c93c21dc1b79c",
    "pipeline": "master"
  },
  {
    "project": "org-meta-hooks",
    "component": "gaia",
    "pr-id": "477",
    "commit-sha": "ed2c269b280a1abfebd04b92940dc2a1356cc86d",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "org-meta-hooks",
    "component": "backend",
    "pr-id": "730",
    "commit-sha": "6f48a9dc03c5303e204c6ef4414aee7eebd9d842",
    "pipeline": "master"
  },
  {
    "project": "org-meta-hooks",
    "component": "gaia",
    "pr-id": "477",
    "commit-sha": "ed2c269b280a1abfebd04b92940dc2a1356cc86d",
    "pipeline": "master"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "org-meta-hooks",
    "component": "backend",
    "pr-id": "730",
    "commit-sha": "7601ce44d30606ae17c50821fe7bf515c6f63ef8",
    "pipeline": "master"
  },
  {
    "project": "org-meta-hooks",
    "component": "gaia",
    "pr-id": "477",
    "commit-sha": "ed2c269b280a1abfebd04b92940dc2a1356cc86d",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "org-meta-hooks",
    "component": "backend",
    "pr-id": "730",
    "commit-sha": "7601ce44d30606ae17c50821fe7bf515c6f63ef8",
    "pipeline": "master"
  },
  {
    "project": "org-meta-hooks",
    "component": "gaia",
    "pr-id": "477",
    "commit-sha": "ed2c269b280a1abfebd04b92940dc2a1356cc86d",
    "pipeline": "master"
  }
]
```

**Pull request merged**